socialstar.rb

```diff
@@ -0,0 +1,263 @@
+class Socialstar
+  include Mongoid::Document
+  include Mongoid::Timestamps
+  include Mongoid::Enum
+
+  devise :omniauthable, :omniauth_providers => [:facebook, :instagram, :twitter, :google_oauth2]
+
+  has_one :invitation_code
+
+  CROWDTAP_ATTRIBUTES = [
+    :country_code,
+    :date_of_birth,
+    :email,
+    :fb_user_id,
+    :first_name,
+    :gender,
+    :last_name,
+    :social_media_access_tokens,
+    :terms_of_service,
+    :ethnicity,
+    :relationship_status,
+    :parenting_status,
+    :zip_code
+  ]
+
+  VALID_SOCIAL_CHANNELS = {
+    :blog      => [:name, :url, :visitors, :payment, :categories => []],
+    :facebook  => [:username, :followers, :payment],
+    :instagram => [:username, :followers, :payment],
+    :pinterest => [:username, :followers, :payment],
+    :tumblr    => [:username, :followers, :payment],
+    :twitter   => [:username, :followers, :payment],
+    :vine      => [:username, :followers, :payment],
+    :youtube   => [:channel, :subscribers, :payment]
+  }
+
+  VALID_SOCIAL_CHANNELS.each do |network, fields|
+    fields.each do |field|
+      next if field == :payment
+      if field.is_a? Hash
+        field = field.keys.first
+      end
+      getter = "#{network}_#{field}".to_sym
+      CROWDTAP_ATTRIBUTES.push getter
+
+      define_method getter do
+        if social_channels[network]
+          social_channels[network][field]
+        end
+      end
+    end
+  end
+
+  US_STATES = ['AL','AK','AZ','AR','CA','CO','CT','DE','FL','GA','HI','ID','IL','IN','IA','KS','KY',
+               'LA','ME','MD','MA','MI','MN','MS','MO','MT','NE','NV','NH','NJ','NM','NY','NC','ND',
```

Comment:
Line is too long. [100/80]<br>Space missing after comma.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.